docs/source/name_inference.rst

```diff
@@ -0,0 +1,426 @@
+.. currentmodule:: torch
+
+.. _name_inference_reference-doc:
+
+Named Tensors operator coverage
+==============================
+
+.. warning::
+    The named tensor API is experimental and subject to change.
+
+Named tensors use names to check that some APIs are being used correctly and
+propagate names in a process called name inference.
+Name inference works as the following:
```

Comment:
maybe "does the following" ? "works as the following" sounds odd to me.

Reply:
works as follows?

Reply:
I'm not the best writer, please keep sending comments of things like this that can be improved.